web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/ThumbnailController.java

```diff
@@ -0,0 +1,114 @@
+package edu.unc.lib.boxc.web.services.rest.modify;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.models.AccessGroupSet;
+import edu.unc.lib.boxc.auth.api.services.AccessControlService;
+import edu.unc.lib.boxc.model.api.ids.PID;
+import edu.unc.lib.boxc.model.api.objects.FileObject;
+import edu.unc.lib.boxc.model.api.objects.RepositoryObjectLoader;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import edu.unc.lib.boxc.operations.jms.thumbnail.ThumbnailRequest;
+import edu.unc.lib.boxc.operations.jms.thumbnail.ThumbnailRequestSender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import edu.unc.lib.boxc.auth.api.models.AgentPrincipals;
+import edu.unc.lib.boxc.auth.fcrepo.models.AgentPrincipalsImpl;
+import edu.unc.lib.boxc.web.services.processing.ImportThumbnailService;
+
+import static edu.unc.lib.boxc.auth.fcrepo.services.GroupsThreadStore.getAgentPrincipals;
+
+/**
+ * Controller for handling thumbnail requests, including:
+ * upload submission for collection display thumbnails,
+ * assigning a child object to use as a thumbnail
+ *
+ * @author lfarrell
+ *
+ */
+@Controller
+public class ThumbnailController {
+    private static final Logger log = LoggerFactory.getLogger(ThumbnailController.class);
+
+    @Autowired
+    private ImportThumbnailService service;
+    @Autowired
+    private ThumbnailRequestSender thumbnailRequestSender;
+    @Autowired
+    private AccessControlService aclService;
+    @Autowired
+    private RepositoryObjectLoader repositoryObjectLoader;
+
+    @PostMapping(value = "edit/displayThumbnail/{pid}")
+    public @ResponseBody
+    ResponseEntity<Object> ImportThumbnail(@PathVariable("pid") String pid,
```

Comment:
I know this isn't new to this pr, but we should change this to importThumbnail since java conventions are for method names to start with the lowercase, which I missed in the original PR.